### PR TITLE
Enforcing atleast NServiceBus v5.2.8, fixes #140

### LINKED ISSUE #140: Using NServiceBus pre 5.2.8 with version 6.2.5 results in NullReferenceException
Using NServiceBus version 5.2.4 with NServiceBus.NHibernate version 6.2.5 results in a NullReferenceException

>Failed to fetch timeouts from the timeout storage
System.NullReferenceException: Object reference not set to an instance of an object.
  at NServiceBus.Timeout.Hosting.Windows.TimeoutPersisterReceiver.Poll(Object obj) in c:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Timeout\Hosting\Windows\TimeoutPersisterReceiver.cs:line 88
  at System.Threading.Tasks.Task.Execute()

It seems that version 6.2.3 part of Particular/NServiceBus#2885 that introduced `IPersistTimeoutsV2` caused this NRE behavior and that the backwards compatibility on the `IPersistTimeouts` interface is not working.

Reason for it not working is that the implementations inherit both `IPersistTimeouts` and `IPersistTimeoutsV2`. If a separate V2 implementation was added there would not be a runtime dependency on `IPersistTimeoutsV2`.

## Comparing to NServiceBus.RavenDB

Looking at package NServiceBus.RavenDB version 3.0.4 that introducted  `IPersistTimeoutsV2` and that is targeting atleast NServiceBus 5.2.8

## Howto reproduce it

- Copy/clone http://docs.particular.net/samples/hosting/nservicebus-host/
- Add latest version of `NServiceBus.NHibernate` package
- Configure it to use NServiceBus.NHibernate
- Run it and see that it works as expected
- Downgrade `NServiceBus` to 5.2.7
- Run it again and soon you see a NRE

Notice that NHibernate is not initialized at all.

All the latest versions of other official transports target at least NServiceBus v5.2.8 which makes NServiceBus to be automatically updated if they add these (latest) transport packages to their project via the nuget management UI and this behavior cannot occur.

However this exception can still occur with other transports too if updating to a specific older version:

* NServiceBus.RavenDB v2.1.2 & NServiceBus v5.2.7
* NServiceBus.Azure v6.1.5 & NServiceBus v5.2.7

## References

- https://github.com/Particular/NServiceBus/issues/2885
- https://github.com/Particular/NServiceBus.NHibernate/releases/tag/6.2.3
- https://github.com/Particular/NServiceBus.RavenDB/releases/tag/3.0.4
- https://www.nuget.org/packages/NServiceBus.NHibernate/6.2.3
- https://www.nuget.org/packages/NServiceBus.RavenDB/3.0.4

### PR DESCRIPTION
**Pull request by ramonsmits**

### CODE REVIEW

**Comment by ramonsmits** [comment]
@andreasohlund @weralabaj Not entirely sure about this release yet. This does for users to update to at least v5.2.8 when running 5.1.x or 5.0

If we do it for NServiceBus v5 then why not 

Alternatives:

- Fix the persistence packages to not have implementations that are dependant on a non existing interface
- Only mention this somewhere

**Comment by SzymonPobiega** [comment]
@ramonsmits can you explain the changes to `OpenSessionBehavior`. I don't understand them. I need to release 6.2.6 in order to fix https://github.com/Particular/NServiceBus.NHibernate/issues/158.

**Comment by ramonsmits** [comment]
@SzymonPobiega I didn't do anything with the code. It only updates the nuget packages dependancies. Similar to that of RavenDB.

**Comment by SzymonPobiega** [comment]
@ramonsmits I've reverted that fix for #142 and fixed it on different level. Take a look when you have some time.

**Comment by SzymonPobiega** [comment]
@SeanFeldman please take a look at this. IMO it is ready to release. Waiting for @ramonsmits to confirm if my change of his change is OK ;-)

**Comment by SeanFeldman** [comment]
@SzymonPobiega :+1: for #158

File: src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs [hunk 1]
```diff
@@ -106,7 +106,7 @@ void PerformCleanup(object state)
                     if (cleanupFailures >= 10)
                     {
                         criticalError.Raise("Failed to clean expired Outbox records after 10 consecutive unsuccessful attempts. The most likely cause of this is connectivity issues with your database.", ex);
-                        throw;
+                        cleanupFailures = 0;
```
Comment (ramonsmits) [inline src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs]: Why are you resetting `cleanupFailures`?

**Comment by ramonsmits** [comment]
@SzymonPobiega Last commit looks good. Just one question, I do know that some lazy things have coupling with the SQL transport. Is the `Lazy<ITransaction>` one of them? Isn't it used for connection sharing?

File: src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs (hunk 1 above)
Comment (SzymonPobiega) [inline src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs]: To not raise critical errors in a tight loop.

**Comment by SzymonPobiega** [comment]
nope, `Lazy<ITransaction>` is used when using native tx (e.g. outbox)

**Comment by ramonsmits** [comment]
@SzymonPobiega Then :+1:

**Pull request merged**